src/app/server/Server.cpp

```diff
@@ -254,45 +254,52 @@ CHIP_ERROR Server::Init(AppDelegate * delegate, uint16_t secureServicePort, uint
     //
     // This is disabled for thread device because the same code is already present for thread devices in
     // src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread_LwIP.cpp
-    // https://github.com/project-chip/connectedhomeip/issues/14254
 #if !CHIP_DEVICE_CONFIG_ENABLE_THREAD
+    RejoinExistingMulticastGroups();
+#endif // !CHIP_DEVICE_CONFIG_ENABLE_THREAD
+
+exit:
+    if (err != CHIP_NO_ERROR)
     {
-        ChipLogProgress(AppServer, "Adding Multicast groups");
-        ConstFabricIterator fabricIterator = mFabrics.cbegin();
-        while (!fabricIterator.IsAtEnd())
-        {
-            const FabricInfo & fabric = *fabricIterator;
-            Credentials::GroupDataProvider::GroupInfo groupInfo;
+        ChipLogError(AppServer, "ERROR setting up transport: %s", ErrorStr(err));
+    }
+    else
+    {
+        ChipLogProgress(AppServer, "Server Listening...");
+    }
+    return err;
+}
+
+void Server::RejoinExistingMulticastGroups()
+{
+    ChipLogProgress(AppServer, "Joining Multicast groups");
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    for (const FabricInfo & fabric : mFabrics)
+    {
+        Credentials::GroupDataProvider::GroupInfo groupInfo;
 
-            Credentials::GroupDataProvider::GroupInfoIterator * iterator =
-                mGroupsProvider.IterateGroupInfo(fabric.GetFabricIndex());
+        auto * iterator = mGroupsProvider.IterateGroupInfo(fabric.GetFabricIndex());
+        if (iterator)
+        {
             while (iterator->Next(groupInfo))
             {
                 err = mTransports.MulticastGroupJoinLeave(
                     Transport::PeerAddress::Multicast(fabric.GetFabricIndex(), groupInfo.group_id), true);
```

Comment:
What does the "true" stand for?

Reply:
True means the `GroupDataProvider` was able to allocate an `GroupInfoIterator`.
I added a comment explaining the true scenario